app/operators/library/ajax/user_actions.php

```diff
@@ -164,22 +157,67 @@
             } else {
                 $already_disabled_enc = htmlspecialchars(implode(", ", $already_disabled), ENT_QUOTES, 'UTF-8');
                 $already_disabled_label = ((count($to_disable) > 1 || count($to_disable) == 0)) ? "users" : "user";
-                
                 $class = "danger";
                 $message = sprintf('%s <strong>%s</strong> already disabled.', $already_disabled_label, $already_disabled_enc);
             }
-
             break;
-        
+//============================
+        case 'userMail':
+            // Prepare the SQL query to retrieve user mail information
+            $sql = sprintf(
+                "SELECT radcheck.username, value, email, firstname, lastname
+        FROM %s
+        JOIN %s
+        ON radcheck.username = userinfo.username
+        WHERE radcheck.username IN (%s);",
+                $configValues['CONFIG_DB_TBL_RADCHECK'], // Table containing user credentials
+                $configValues['CONFIG_DB_TBL_DALOUSERINFO'], // Table containing user info
+                $username_list // List of usernames to filter the results
+            );
+
+            // Execute the SQL query
+            $res = $dbSocket->query($sql);
+
+            // Iterate through the results
+            while ($row = $res->fetchRow()) {
+                // Get the recipient's email address and username
+                $recipient_email_address = $row[2]; // Email of the user
+                $recipient_name = $row[0]; // Username of the user
+
+                // Set the subject and body of the email
+                $subject = 'VPN Credentials'; // Subject of the email
+                $body = sprintf(
+                    '<b>VPN credential</b><br>Hello, %s %s!<br>Your login is: %s<br>Your password is: %s<br>VPN Server is: vpn.company.com<br><br>Best regards, Admin',
```

Comment:
what is the `vpn.company.com` for?

Could you make that variable configurable from the $configValues['...'] and put a placeholder in the config files so others can easily change it?

Reply:
Done.